SharpSource/SharpSource/Diagnostics/SwitchDoesNotHandleAllEnumOptionsAnalyzer.cs

```diff
@@ -27,35 +25,29 @@ public override void Initialize(AnalysisContext context)
     {
         context.EnableConcurrentExecution();
         context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-        context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.SwitchStatement);
+        context.RegisterOperationAction(AnalyzeSwitchOperation, OperationKind.Switch);
     }
 
-    private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+    private static void AnalyzeSwitchOperation(OperationAnalysisContext context)
     {
-        var switchBlock = (SwitchStatementSyntax)context.Node;
-
-        if (context.SemanticModel.GetTypeInfo(switchBlock.Expression).Type is not INamedTypeSymbol enumType || enumType.TypeKind != TypeKind.Enum)
+        var switchOperation = (ISwitchOperation)context.Operation;
+        if (switchOperation.Value.Type is not INamedTypeSymbol { TypeKind: TypeKind.Enum } enumType)
         {
             return;
         }
 
         var labelSymbols = new HashSet<ISymbol>(SymbolEqualityComparer.Default);
-        foreach (var section in switchBlock.Sections)
+        foreach (var caseOperation in switchOperation.Cases)
         {
-            foreach (var label in section.Labels)
+            foreach (var caseClauseOperation in caseOperation.Clauses)
             {
-                if (label.IsKind(SyntaxKind.CaseSwitchLabel))
+                if (caseClauseOperation is ISingleValueCaseClauseOperation { Value: IFieldReferenceOperation enumReference })
                 {
-                    var switchLabel = (CaseSwitchLabelSyntax)label;
-                    var symbol = context.SemanticModel.GetSymbolInfo(switchLabel.Value).Symbol;
```

Comment:
I'm suspecting `GetSymbolInfo` call here is expensive and is the reason of the performance issue.